### PR TITLE
Feat/e2e custom card

### PR DESCRIPTION
**Pull request by Yanhuazong**
Add test to custom card block.

### CODE REVIEW

File: src/card/block.spec.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    test( '🔬 Block should be available.', async () => {
+        await insertBlock( 'Custom Card' )
+
+        // tests that the block is properly inserted and matches the existing snapshot
+        expect(await page.$('[data-type="purdue-blocks/card"]')).not.toBeNull()
+        expect( await getEditedPostContent()).toMatchSnapshot()
+    } )
+
+    describe( '🔬 Block Editor Fields', () => {
+        test( '🔬 Button should open Media Library.', async () => {
```
Comment (jyoung295) [inline src/card/block.spec.js:36]: Just put the microscope on the describes, not all the tests.

File: src/card/block.spec.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+                await clickCheckbox('Add a link to this card?')
+        
+                const linkInputs = await page.$x(
+                    `//label[@class = "components-checkbox-control__label"][.="Call to action text" or .="Link address" or .="Open link in new tab?"]`
+                )
+        
+                // check that the checkboxes appear and that the correct number exist, 3
+                expect(linkInputs).not.toBeNull()
+                expect(linkInputs.length).toBe(1)
```
Comment (jyoung295) [inline src/card/block.spec.js:138]: Shouldn't this be testing for three items in the array? 

`expect(linkInputs.length).toBe(3)`

File: src/card/block.spec.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+            // expect( editedContent).toMatchSnapshot()
+        })
+        describe( '🔬 Link can be added to the card', () => {
+            test( '🔬 "Add a link to this card?" checkbox reveals link inputs', async () => {
+                await blockStartup(block)
+
+                await clickCheckbox('Add a link to this card?')
+        
+                const linkInputs = await page.$x(
+                    `//label[@class = "components-checkbox-control__label"][.="Call to action text" or .="Link address" or .="Open link in new tab?"]`
```
Comment (jyoung295) [inline src/card/block.spec.js:133]: this is only selecting the checkbox control label, not the other 2 textboxes that show up.

Comment (jyoung295) [reply src/card/block.spec.js:133]: You could change the XPath query to just `//label[.="Call to action text" or .="Link address" or .="Open link in new tab?"]`
That way it avoids looking for a label with a specific class, and only looks for labels with those 3 text options.

**Review by jyoung295 — CHANGES_REQUESTED** [review]
Please address my comments and then re-request review. Great start though, thanks!

File: src/card/block.spec.js (hunk 1 above)
Comment (Yanhuazong) [reply src/card/block.spec.js:36]: I found it was still a little hard to read in the snapshot so I added them.

Comment (jyoung295) [reply src/card/block.spec.js:36]: Yeah you're not wrong, maybe we should find a different one for the tests just to differentiate

File: src/card/block.spec.js (hunk 3 above)
Comment (Yanhuazong) [reply src/card/block.spec.js]: Ah, I thought it also tests the show-up of the text boxes at the  beginning and later I forgot to update it.

File: src/card/block.spec.js (hunk 2 above)
Comment (Yanhuazong) [reply src/card/block.spec.js]: It was a mistake. Updated it in a new commit.

File: src/card/block.spec.js [hunk 4]
```diff
@@ -128,14 +128,20 @@ describe( '🔬 Card Block', () => {
...
+                
         
-                const linkInputs = await page.$x(
-                    `//label[@class = "components-checkbox-control__label"][.="Call to action text" or .="Link address" or .="Open link in new tab?"]`
-                )
-        
-                // check that the checkboxes appear and that the correct number exist, 3
-                expect(linkInputs).not.toBeNull()
-                expect(linkInputs.length).toBe(1)
+                // click to uncheck the checkbox, outline should now turn off
```
Comment (jyoung295) [inline src/card/block.spec.js:138]: remove this comment that was copy pasted 😉

Comment (Yanhuazong) [reply src/card/block.spec.js]: Done. Thanks! :)

File: src/card/block.spec.js (hunk 1 above)
Comment (jyoung295) [reply src/card/block.spec.js:36]: what about a magnifying glass 🔎

Comment (jyoung295) [reply src/card/block.spec.js:36]: or a page 📄

Comment (jyoung295) [reply src/card/block.spec.js:36]: or maybe a test tube 🧪

Comment (Yanhuazong) [reply src/card/block.spec.js:36]: I think all of them work. Maybe we just pick 🔎

Comment (jyoung295) [reply src/card/block.spec.js:36]: Haha okay go ahead and use that one and I can go back through the tests I've already made and add it

Thanks

Comment (Yanhuazong) [reply src/card/block.spec.js]: Done. Thanks!

**Pull request merged**